### PR TITLE
Added the Swift retrieval metrics

### PR DESCRIPTION
**Pull request by jlarriba**

### CODE REVIEW

File: controllers/ceilometer_controller.go [hunk 1]
```diff
@@ -854,3 +859,38 @@ func (r *CeilometerReconciler) findObjectsForSrc(ctx context.Context, src client
...
+	if err != nil || (ctrlResult != ctrl.Result{}) {
+		return ctrlResult, err
+	}
+
+	// We are using the fixed domainID "default" because it is also fixed in ceilometer.conf
+	user, err := os.GetUser(log, instance.Spec.ServiceUser, "default")
+
+	err = os.AssignUserDomainRole(
+		log,
+		"SwiftSystemReader",
```
Comment (csibbitt) [inline controllers/ceilometer_controller.go:888]: A much nicer role name than "ResellerAdmin" :)

**Review by csibbitt — APPROVED** [review]

**Comment by jlarriba** [comment]
recheck

**Comment by vyzigold** [comment]
```
2024-03-22T11:32:05Z	ERROR	Reconciler error	{"controller": "ceilometer", "controllerGroup": "telemetry.openstack.org", "controllerKind": "Ceilometer", "Ceilometer": {"name":"ceilometer","namespace":"telemetry-kuttl-tests"}, "namespace": "telemetry-kuttl-tests", "name": "ceilometer", "reconcileID": "8112d651-948f-4f84-86f4-8a53af097dae", "error": "SwiftSystemReader role not found in keystone"}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/controller/controller.go:329
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/controller/controller.go:266
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/controller/controller.go:227
```

**Comment by jlarriba** [comment]
Good catch in Kuttl here. It is true that the SwiftSystemReader role cannot exist on Kuttl because we do not deploy the Swift operator. But we cannot be dependendant on the Swift operator existence to run, so I am going to modify this to search for the SwiftSystemRole. If it exists, we add it to Ceilometer, if it does not, we do nothing and continue without any issues.

**Comment by jlarriba** [comment]
recheck

_error: the cert-manager CRDs are not yet installed on the Kubernetes API server_

**Comment by jlarriba** [comment]
recheck

File: config/manager/kustomization.yaml [hunk 2]
```diff
@@ -12,5 +12,4 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: controller
-  newName: quay.io/openstack-k8s-operators/telemetry-operator
-  newTag: latest
+  newName: quay.io/jlarriba/telemetry-operator-bundle
```
Comment (vyzigold) [inline config/manager/kustomization.yaml:15]: I'm not sure about where this came from, but shouldn't this be left to "openstack-k8s-operators"?

Comment (jlarriba) [reply config/manager/kustomization.yaml:15]: Aw, this is a remain of my local testing

File: controllers/ceilometer_controller.go (hunk 1 above)
Comment (jlarriba) [reply controllers/ceilometer_controller.go:930]: Yes indeed.

**Pull request merged**